pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -293,10 +314,16 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		)
 		return configMetaErr
 	}
-
-	if state == clusterMembersChangedUnready || (cr.Spec.ConnectionsGenToProcess != cr.Status.LastConnectionGen) {
-		cr.Status.SpecGenerationToProcess = &cr.Generation
-		cr.Status.LastConnectionGen = cr.Spec.ConnectionsGenToProcess
+	newHash := calcConnectionsHash(&cr.Spec.Connections, cr.Namespace)
```

Comment:
If the cluster is in stable state we already calculated the hash above. I think we should try to avoid redundantly calculating the hash if possible.

We could for example cache the hash in the cluster CR, much like we cache the app CR in the cluster CR. Cf. in the definition of the KubeDirectorCluster struct type we have the field AppSpec that is flagged to keep it out of the JSON, so it's for internal use only. Whenever we want to get the app for this cluster we call catalog.GetApp which returns the cached app if available and if not finds it and caches it. Could do something very similar with this hash.

(...actually, looking at GetApp now I think that a change a while back broke the caching behavior there. Hmm, and d'oh.)

Reply:
Or if these are the only two places we use the hash we don't need to get that fancy, should just put it in a local variable.